### PR TITLE
docs(readme): added types in readme

### PR DESCRIPTION
**Pull request by harish-sethuraman**
This PR contains a:

- [ ] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [x] **metadata update**

### Motivation / Use-Case
Added proper type definitions in readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -612,14 +647,19 @@ module.exports = {
 
 ### `implementation`
 
-Type: `Function | String`
+Type:
+
+```ts
+type implementation = require('postcss') | string
```
Comment (alexander-akait) [inline README.md:653]: It can be function, let's improve this, `require('postcss')` is default value

Comment (harish-sethuraman) [reply README.md:653]: `require('postcss')` exports a type file called `postcss.d.ts`. That is why I gave `require('postcss')`

Comment (harish-sethuraman) [reply README.md:653]: 🤔  There is no type for package? Shall I leave it generic as `object`?

Comment (alexander-akait) [reply README.md:653]: I think we can do it, just add comment about it should be have the same API as postcss

Comment (harish-sethuraman) [inline README.md:653]:
```suggestion
type implementation = object
```

File: README.md [hunk 2]
```diff
@@ -612,14 +647,19 @@ module.exports = {
...
+```
+
```
Comment (harish-sethuraman) [inline README.md:655]:
```suggestion
type of implementation should be the same as [postcss.d.ts](https://github.com/postcss/postcss/blob/main/lib/postcss.d.ts)
```

Comment (harish-sethuraman) [reply README.md:655]: Is this okay ?

Comment (alexander-akait) [reply README.md:655]: :+1:

**Pull request merged**